feature: filter IdP retrival

strehle opened the pull request:
- New: Filter IdP list by origin

Many calls to IdPs first retrieve all and then filter by origin on client side. The origin has an index in DB, therefore there should be an option to get the IdP by origin filter.

NEW 

Filter: uaac curl -b /identity-providers?origin=sap.proxy | jq -r .[0]

Allow to get IdP based on origin. Do not iterate over all.

Sonar
https://sonarcloud.io/summary/new_code?id=cloudfoundry-identity-parent&pullRequest=2882

strehle committed:
WIP: idp secret

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/idp-secret

strehle committed:
feature: delete secret on existing IdP

- allow to delete a relyingPartySecret on IdP
- Filter IdP list by origin
- Return the auth_method to show current configured client authentication method

strehle committed:
Documentation

strehle committed:
fix names

strehle commented:
generated documentation can be downloaded as ZIP
https://github.com/cloudfoundry/uaa/actions/runs/9034861916?pr=2882

strehle committed:
sonar

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/idp-secret

strehle committed:
sonar

strehle commented:
@torsten-sap 

NEW 

Filter: uaac curl -b /identity-providers?origin=sap.proxy | jq -r .[0]

Allow to get IdP based on origin. Do not iterate over all.

torsten-sap commented:
@strehle In comparison to https://docs.cloudfoundry.org/api/uaa/version/77.1.0/index.html#change-secret, does it make sense to also allow the update here (besides delete)? I would assume this makes things easier for some customers if only the secret shall be changed (although this is of course already possible via update of the IdP).

strehle committed:
Add patch call to change a secret from an external IdP

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/idp-secret

strehle commented:
> @strehle In comparison to https://docs.cloudfoundry.org/api/uaa/version/77.1.0/index.html#change-secret, does it make sense to also allow the update here (besides delete)? I would assume this makes things easier for some customers if only the secret shall be changed (although this is of course already possible via update of the IdP).

Ok, we can support the easy change of a secret, however I would stay with separat HTTP methods, e.g. DELETE and/or PATCH instead of a mix in PUT

See https://github.com/cloudfoundry/uaa/pull/2882/commits/cd32e6149374461b837a21daf02af151ab1361f5 which allows to patch a secret only

strehle commented:
> @strehle In comparison to https://docs.cloudfoundry.org/api/uaa/version/77.1.0/index.html#change-secret, does it make sense to also allow the update here (besides delete)? I would assume this makes things easier for some customers if only the secret shall be changed (although this is of course already possible via update of the IdP).

Created https://github.com/cloudfoundry/uaa/pull/2885 because from Prio the deletion should be added soon, all other changes in REST API of IdPs are an improvement but not a bug fixing

strehle committed:
Alias handling

strehle committed:
Sonar refactorings

See
https://sonarcloud.io/code?id=cloudfoundry-identity-parent&selected=cloudfoundry-identity-parent%3Aserver%2Fsrc%2Fmain%2Fjava%2Forg%2Fcloudfoundry%2Fidentity%2Fuaa%2Fprovider%2FIdentityProviderEndpoints.java

strehle committed:
remove dead code

strehle committed:
more sonar smell fixes

strehle committed:
no log change

strehle committed:
revert and allow test endpoint again

strehle committed:
warnings

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/idp-secret

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into refactorIdp2

# Conflicts:
#	server/src/main/java/org/cloudfoundry/identity/uaa/provider/IdentityProviderEndpoints.java

strehle committed:
rebase

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into refactorIdp2

strehle committed:
rebase

strehle committed:
test setup

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into refactorIdp2

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/refactorIdp2' into feature/idp-secret

# Conflicts:
#	model/src/main/java/org/cloudfoundry/identity/uaa/constants/ClientAuthentication.java
#	model/src/main/java/org/cloudfoundry/identity/uaa/provider/AbstractExternalOAuthIdentityProviderDefinition.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/provider/IdentityProviderEndpoints.java

strehle committed:
rebase

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/idp-secret

# Conflicts:
#	server/src/main/java/org/cloudfoundry/identity/uaa/provider/IdentityProviderEndpoints.java

peterhaochen47 commented on uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointDocs.java:
```diff
@@ -967,6 +967,47 @@ void getAllIdentityProviders() throws Exception {
+                requestParameters(
+                    parameterWithName("rawConfig").optional("false").type(BOOLEAN).description("Flag indicating whether the response should use raw, unescaped JSON for the `config` field of the IDP, rather than the default behavior of encoding the JSON as a string."),
+                    parameterWithName("active_only").optional("false").type(BOOLEAN).description("Flag indicating whether only active IdPs should be returned or all."),
+                    parameterWithName("origin").optional(null).type(STRING).description("<small><mark>UAA 77.10.0</mark></small> Return only IdPs with specific origin.")
```
Is this `origin` the same thing as the `originKey` when you [create the IDPs](https://docs.cloudfoundry.org/api/uaa/version/77.8.0/#saml)? If so, why not just call this field `originKey` to keep it consistent?

strehle replied:
ok done

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/idp-secret

strehle committed:
review
